adapters/repos/db/index_queue.go

```diff
@@ -244,6 +244,23 @@ func (q *IndexQueue) Push(ctx context.Context, vectors ...vectorDescriptor) erro
 	now := time.Now()
 	q.lastPushed.Store(&now)
 
+	// ensure the vector length is the same
+	var dims int
+	for i := range vectors {
```

Comment:
hnsw has a `ValidateBeforeInsert` method - can we use that? Or save the respective length in the indexqueue?

Reply:
This one would only notice inconsistent vector length within one batch, right? If I send objects with length 5 in one batch and then with length 6 in the next one, this would not detect it?

Reply:
> This one would only notice inconsistent vector length within one batch, right? If I send objects with length 5 in one batch and then with length 6 in the next one, this would not detect it?

Exactly, I raised the concern yesterday with the team. It seems better to detect the vector length of the first ever vector and ensure it's consistent with everything else.
I thought it would be a bit more complex as we'd need to either store that information somewhere or pass it to the queue, but I wasn't aware of `ValidateBeforeInsert`. That method definitely solves the problem!

Reply:
I opted for a combination of both keeping track of the dimensions in the queue (volatile) and calling `ValidateBeforeInsert`, because for the first batch the index doesn't yet know the vector length and thus calling `ValidateBeforeInsert` doesn't really do anything.